lms/static/js/header/header.js

```diff
@@ -0,0 +1,172 @@
+            $userMenu.attr('aria-expanded', 'false');
+        } else {
+            $dropdownMenu.show();
+            $dropdownMenu.find('.dropdown-item')[0].focus();
+            $userMenu.attr('aria-expanded', 'true');
+        }
+        $('.toggle-user-dropdown').toggleClass('open');
+    });
+
+    // Toggling menu visibility with the hamburger menu
+    $('.hamburger-menu').click(function() {
+        $hamburgerMenu = $('.hamburger-menu');
+        $mobileMenu = $('.mobile-menu');
+        if ($mobileMenu.is(':visible')) {
+            $mobileMenu.hide();
+            $hamburgerMenu.attr('aria-expanded', 'false');
+        } else {
+            $mobileMenu.show();
+            $hamburgerMenu.attr('aria-expanded', 'true');
+        }
+        $hamburgerMenu.toggleClass('open');
+    });
+
+    // Hide hamburger menu if no nav items (sign in and register pages)
+    if ($('.mobile-nav-item').size() === 0) {
+        $('.hamburger-menu').css('display', 'none');
+    }
+
+    createMobileMenu();
+});
+
+// Ensure click away hides the user dropdown
+$(window).click(function(e) {
+    'use strict';
+    if (!$(e.target).is('.dropdown-item, .toggle-user-dropdown')) {
+        $('.global-header .nav-item .dropdown-user-menu').hide();
+    }
+});
+
+// Accessibility keyboard controls for user dropdown and mobile menu
+$(document).on('keydown', function(e) {
+    'use strict';
+    var isNext;
+    var nextLink;
+    var loopFirst;
+    var loopLast;
+    var isLastItem = $(e.target).parent().is(':last-child');
+    var isToggle = $(e.target).hasClass('toggle-user-dropdown');
+    var isHamburgerMenu = $(e.target).hasClass('hamburger-menu');
+    var isMobileOption = $(e.target).parent().hasClass('mobile-nav-link');
+    var isDropdownOption = !isMobileOption && $(e.target).parent().hasClass('dropdown-item');
+    var $userMenu = $('.user-dropdown');
+    var $hamburgerMenu = $('.hamburger-menu');
+    var $toggleUserDropdown = $('.toggle-user-dropdown');
+
+    // Open or close relevant menu on enter or space click and focus on first element.
+    if ((e.keyCode === 13 || e.keyCode === 32) && (isToggle || isHamburgerMenu)) {
+        $(e.target).click();
+        if (isHamburgerMenu) {
+            if ($('.mobile-menu').is(':visible')) {
+                $hamburgerMenu.attr('aria-expanded', true);
+                $('.mobile-menu .mobile-nav-link a').first().focus();
+            } else {
+                $hamburgerMenu.attr('aria-expanded', false);
+            }
+        } else if (isToggle) {
+            if ($('.global-header .nav-item .dropdown-user-menu').is(':visible')) {
+                $userMenu.attr('aria-expanded', 'true');
+                $('.dropdown-item a:first').focus();
+            } else {
+                $userMenu.attr('aria-expanded', false);
+            }
+        }
+        // Don't allow for double click or page jump on Firefox browser
+        e.preventDefault();
+        e.stopPropagation();
```

Comment:
Move these to the top of the if statement wrapper, this may be what's causing the inconsistent behaviour.